### ISSUE TITLE
Documentation for using supervised learning 

### ISSUE THREAD

**Issue by munirfarzeen** [BasBuller/PySNN#30]
hi, 
The examples you have given in the docs all represent unsupervised training.
I wonder how i can do supervised training?
What would be loss function?
And how to encode image frame to input to the spiking network?

**Comment by Huizerd** [comment]
Most often backpropagation is used for that. For SNNs, that means we have to find a surrogate derivative of the spiking function. Multiple works exist on this, like https://arxiv.org/pdf/1706.02609.pdf and https://arxiv.org/pdf/1901.09948.pdf. For an implementation in PyTorch, have a look at [Norse](https://github.com/norse/norse). PySNN does not yet fully offer gradient-based learning.

As for encoding, you can do different things. Just google for “spike encoding images”. One example would be to input a sequence of 5 spikes per pixel, where [1,1,1,1,1] would correspond to 255 and [0,0,0,0,0] to 0, giving you an input tensor of (10, 10, 5) for a 10x10 image.
